pyccel/codegen/printing/fcode.py

```diff
@@ -2536,6 +2546,14 @@ def _print_FunctionCall(self, expr):
 
 #=======================================================================================
 
+    def _print_Application(self, expr):
+        if isinstance(expr, PyccelAllocatableObject):
+            errors.report(FORTRAN_ALLOCATABLE_IN_EXPRESSION,
+                          symbol=expr, severity='fatal')
```

Comment:
Can you please explain when this error is raised?

Reply:
This is related to issue #307. It is raised when trying to translate code such as:
```
a = np.ones(4)*5
```